src/core/xfa/template.js

```diff
@@ -615,7 +615,14 @@ class BooleanElement extends Option01 {
   }
 
   [$toHTML](availableSpace) {
-    return HTMLResult.success(this[$content] === 1);
+    return HTMLResult.success({
+      name: "span",
+      attributes: {
+        class: ["xfaRich"],
+        style: {},
+      },
+      value: this[$content] === 1,
+    });
```

Comment:
this code is all very similar between the different objects, what if you moved it to a base class and made all classes inherit this method?
or even just extract the code to a method and call it in all places